test/functional/p2p_quorum_connect.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The PIVX Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php.
+"""Test MN quorum connection flows"""
+
+import time
+
+from test_framework.test_framework import PivxTestFramework
+from test_framework.mininode import P2PInterface
+from test_framework.messages import msg_version
+from random import getrandbits
+from test_framework.util import (
+    assert_equal,
+    assert_true,
+    bytes_to_hex_str,
+    connect_nodes_clique,
+    hash256,
+    hex_str_to_bytes,
+    wait_until,
+)
+
+class TestP2PConn(P2PInterface):
+    def on_version(self, message):
+        pass
+
+class DMNConnectionTest(PivxTestFramework):
+
+    def set_test_params(self):
+        # 1 miner, 1 controller, 6 remote dmns
+        self.num_nodes = 8
+        self.minerPos = 0
+        self.controllerPos = 1
+        self.setup_clean_chain = True
+        self.extra_args = [["-nuparams=v5_shield:1", "-nuparams=v6_evo:101"]] * self.num_nodes
+        self.extra_args[0].append("-sporkkey=932HEevBSujW2ud7RfB1YF91AFygbBRQj3de3LyaCRqNzKKgWXi")
+
+    def add_new_dmn(self, mns, strType, op_keys=None, from_out=None):
+        mn = self.register_new_dmn(2 + len(mns),
+                                   self.minerPos,
+                                   self.controllerPos,
+                                   strType,
+                                   outpoint=from_out,
+                                   op_blskeys=op_keys)
+        mns.append(mn)
+        return mn
+
+    def check_mn_enabled_count(self, enabled, total):
+        for node in self.nodes:
+            node_count = node.getmasternodecount()
+            assert_equal(node_count['enabled'], enabled)
+            assert_equal(node_count['total'], total)
+
+    def wait_until_mnsync_completed(self):
+        SYNC_FINISHED = [999] * self.num_nodes
+        synced = [-1] * self.num_nodes
+        timeout = time.time() + 160
```

Comment:
i had some pretty inconsistent results with hitting this timeout locally, an increase to `240` seemed to clear it up